src/graph/entities/node.c

```diff
@@ -53,6 +53,45 @@ Node *Node_Clone(const Node *n) {
 	return clone;
 }
 
+int Node_ToString(const Node *n, char *buffer, int bufferLen, GraphEntityStringFromat format) {
+	if(bufferLen <= 1) return 0;
+	int bytes_written = snprintf(buffer, bufferLen, "(");
+	bufferLen -= bytes_written;
+	int currentWriteLength = 0;
+
+	// write id
+	if(format & ENTITY_ID) {
+		currentWriteLength = snprintf(buffer + bytes_written, bufferLen, "%llu", ENTITY_GET_ID(n));
+		bytes_written += currentWriteLength;
+		bufferLen -= currentWriteLength;
+	}
+
+	// write label
+	if(bufferLen > 2 && format & ENTITY_LABELS_OR_RELATIONS) {
```

Comment:
Why dio you check for `bufferLen` here unlike previous condition,
isn't snprintf suppose to make sure we're not overflowing?

Reply:
no 
from [here](http://www.cplusplus.com/reference/cstdio/snprintf/)
> The number of characters that would have been written if n had been sufficiently large, not counting the terminating null character.